app/views/agents/show.html.erb

```diff
@@ -161,17 +161,20 @@
               <b>Options:</b>
               <pre><%= Utils.pretty_jsonify @agent.options || {} %></pre>
             </p>
-
-            <p id="memory" data-agent-id="<%= @agent.id %>">
-              <b>Memory:</b>
-              <% if @agent.memory.present? %>
-                <i class="fa fa-spinner fa-pulse spinner"></i>
-                <i class="fa fa-trash action-icon clear"></i>
-              <% end %>
-              <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
-            </p>
+            <button type="submit" class="btn btn-default" id="toggle-memory">Show Memory</button>
+              <div id="memorypanel" class="hidden">
+                <p id="memory" data-agent-id="<%= @agent.id %>">
+                  <b>Memory:</b>
+                  <% if @agent.memory.present? %>
+                    <i class="fa fa-spinner fa-pulse spinner"></i>
+                    <i class="fa fa-trash action-icon clear"></i>
+                  <% end %>
+                    <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
+                </p>
+              </div>
+            </div>
           </div>
         </div>
       </div>
   </div>
-</div>
+</div>
```

Comment:
It is [good practice](https://robots.thoughtbot.com/no-newline-at-end-of-file) to have an empty new line at the end of every file.